app/components/AddOrganisationFacility.tsx

```diff
@@ -0,0 +1,125 @@
+import React, {useState} from 'react';
+import {Button, Modal} from 'react-bootstrap';
+import JsonSchemaForm, {IChangeEvent} from 'react-jsonschema-form';
+import FormFieldTemplate from 'containers/Forms/FormFieldTemplate';
+import FormObjectFieldTemplate from 'containers/Forms/FormObjectFieldTemplate';
+import addOperatorSchema from 'components/organisation/addOrganisationSchema';
+import addFacilitySchema from 'components/facility/addFacilitySchema';
+
+interface Props {
+  onAddOrganisation?: (...args: any[]) => void;
+  onAddFacility?: (...args: any[]) => void;
+  organisationRowId?: number;
+  reportingYear?: number;
```

Comment:
Aren't all of those props required?

Reply:
They are required conditionally, if AddOrganisationFacility is being rendered from the FacilityList container it needs the bottom 3 props, if being rendered from the Organisations container it only needs the first prop.